htlcswitch/failure_detail.go

```diff
@@ -0,0 +1,60 @@
+package htlcswitch
+
+import (
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/lnwire"
+
+	"github.com/lightningnetwork/lnd/lntypes"
+)
+
+// FailureDetail is an interface which is used to enrich failures with
+// additional information.
+type FailureDetail interface {
+	// FailureString returns a string representation of a failure detail.
+	FailureString() string
```

Comment:
is this called anywhere in this pr?

a more concise name might be `Failure()` or `Detail()`, but use whatever you prefer :)

Reply:
Didn't want to have an empty interface so just picked something that seemed unique :)

This isn't used (could have been used in the `Error` funcs) but since we don't really need these strings and the metadata isn't used anywhere I'm switching these over to an enum. for now. If we end up needing metadata, it'll be easy to switch back.